caikit_nlp/resources/pretrained_model/hf_auto_causal_lm.py

```diff
@@ -177,13 +178,84 @@ def _get_data_collator(self, **kwargs) -> "transformers.DataCollator":
             tokenizer=self._tokenizer, return_tensors="pt", **collator_kwargs
         )
 
+    ### Tokenization strategy implementations
+    # Chunked causal language modeling
+    @classmethod
+    def _causal_lm_as_chunked(
+        cls,
+        tokenizer: "AutoTokenizer",
+        source: str,
+        target: str,
+        max_source_length: int,
+        max_target_length: int,
+        batched_mode: bool,
+        task_ids: Union[None, int],
+        chunk_size: int,
+        drop_remainder: bool,
+    ) -> Union[DataStream[BatchEncoding], BatchEncoding]:
+        """Given a source and target string, build the chunked concatenated sequence and formulate
+        the batch encoded chunks for the sequence. If running in batch mode, the chunks will be
+        collapsed into a single batch encoding for the whole sequence. Otherwise, each chunk will
+        placed in its own BatchEncoding and encapsulated within a datastream.
+
+        Args:
+            tokenizer: AutoTokenizer
+                Tokenizer object to be applied to input records.
+            source: str
+                Raw source string.
+            target: str
+                Raw target string.
+            max_source_length: int
+                Maximum length for input sequences.
+            max_target_length: int
+                Maximum length for output sequences.
+            batched_mode: bool
+                Whether or not we should produce a stream of encodings or a single
+                encoding representing all of the chunked sequence.
+            task_ids: Union[None, int]
+                Task IDs to be used for multiprompt tuning.
+            chunk_size: int
+                unsigned int value to be used for chunk size.
+            drop_remainder: bool
+                Whether or not to keep the residual as an extra chunk if the
+                total number of tokens is not divisible by the chunk size.
+
+        Returns:
+            Union[DataStream[BatchEncoding], BatchEncoding]
+                Encoded chunked sequence as a stream or batch encoding object.
+        """
+        source_ids = tokenizer(source, max_length=max_source_length, truncation=True)
+        target_ids = tokenizer(target, max_length=max_target_length, truncation=True)
+
+        # Force everything to a list of batch encodings; for non-batch mode, this just
+        # puts it into a list. For batch mode, we get a list of batch encodings,
+        # allowing us to standardize subsequent processing a bit.
+        source_id_chunks = cls._force_to_batch_encoding_list_of_chunks(
```

Comment:
nit: for readability, can we add an example in comments of how the `source_id_chunks` would look like